velox/functions/prestosql/StringFunctions.h

```diff
@@ -255,6 +255,13 @@ struct SubstrFunction {
     if (start < 0) {
       start = numCharacters + start + 1;
     }
+    
+    // Following vanilla spark semantics. When the start is 0 after adjusing start,
+    // Keep the origin input. 
+    if (start <= 0) {
+      start = 1;
+      length = numCharacters;
+    }
```

Comment:
Maybe a final solution is to implement this function under sparksql folder.

Reply:
Does Presto has such UT? If presto use the same logic, then we can modify from here directly. If Presto doesn't have the check, we should copy the function to sparksql folder

Reply:
Presto doesn't have this check. And I will put this function into sparksql folder later.

Reply:
we may submit an issue to Velox firstly and ask if presto should have this check or not before we put it in sparksql function.

Reply:
Presto use the following check and will return empty result, which is different with vanilla spark.

 ```
    // Following Presto semantics
    if (start <= 0 || start > numCharacters || length <= 0) {
      result.setEmpty();
      return true;
    }
```

Reply:
I see. In this way we must re-implement the function in sparksql

Reply:
Yes. I will file a new PR to add this function in sparksql.